### PR TITLE
.Net: Openai V2 - IT Passing, Client Request Body Optionality Improvements.

### PR DESCRIPTION
**Pull request by RogerBarreto**
### Motivation and Context

- Reduce unnecessary request body information when not explicitly provided by the caller.
- Ensure all Integration Tests passes 
- Ensure all Unit Tests passes
-

### CODE REVIEW

File: dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs [hunk 1]
```diff
@@ -181,12 +181,12 @@ internal async Task<IReadOnlyList<ChatMessageContent>> GetChatMessageContentsAsy
...
 
             // If we don't want to attempt to invoke any functions, just return the result.
-            if (!toolCallingConfig.AutoInvoke)
+            if (!(toolCallingConfig?.AutoInvoke ?? false))
             {
                 return [chatMessageContent];
             }
 
-            Debug.Assert(kernel is not null);
+            // Debug.Assert(kernel is not null);
```
Comment (SergeyMenshykh) [inline dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs:189]: Is this change intentional?

**Review by SergeyMenshykh — APPROVED** [review]

File: dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs (hunk 1 above)
Comment (RogerBarret0) [reply dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs:189]: Leaving this was breaking my integration tests (breaking the test runtime)

**Review by markwallace-microsoft — CHANGES_REQUESTED** [review]

File: dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs [hunk 2]
```diff
@@ -1134,11 +1143,11 @@ await autoFunctionInvocationFilters[index].OnAutoFunctionInvocationAsync(context
         }
     }
 
-    private ToolCallingConfig GetToolCallingConfiguration(Kernel? kernel, OpenAIPromptExecutionSettings executionSettings, int requestIndex)
+    private ToolCallingConfig? GetToolCallingConfiguration(Kernel? kernel, OpenAIPromptExecutionSettings executionSettings, int requestIndex)
     {
         if (executionSettings.ToolCallBehavior is null)
         {
-            return new ToolCallingConfig(Tools: [s_nonInvocableFunctionTool], Choice: ChatToolChoice.None, AutoInvoke: false);
+            return null;
```
[the remark this answers is not in the document]
Comment (RogerBarreto) [reply dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs:1150]: This is internal behavior, we can change freely, I will make this change, but my logical thinking here is, if you don't want auto function invocation whatsoever any config is nullified (config is not set).

File: dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs [hunk 3]
```diff
@@ -1138,7 +1142,7 @@ private ToolCallingConfig GetToolCallingConfiguration(Kernel? kernel, OpenAIProm
     {
         if (executionSettings.ToolCallBehavior is null)
         {
-            return new ToolCallingConfig(Tools: [s_nonInvocableFunctionTool], Choice: ChatToolChoice.None, AutoInvoke: false);
+            return new ToolCallingConfig(Tools: null, Choice: null, AutoInvoke: false);
```
Comment (markwallace-microsoft) [inline dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs:1145]: Need to update the record so that `Choice` is nullable

**Review by markwallace-microsoft — APPROVED** [review]
